### ISSUE TITLE
Dropout should give an error message when rate is greater than 1

### ISSUE THREAD

**Issue by kmader** [keras-team/keras#10629]
It is a minor issue but dropout should throw a warning or an error if the rate is greater than 1 (it already caps it)

https://github.com/keras-team/keras/blob/master/keras/layers/core.py#L101-L106

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by kmader** [comment]
For background I had a nasty little typo in a model where instead of having `Dense(256)`, I had  `Dropout(256)` which doesn't make sense and results in a model learning nothing because 100% of the values are being set to 0 at training time. It's probably not a common problem but a warning would be useful
